zephyr/Kconfig

```
@@ -0,0 +1,137 @@
+	help
+		Enables Zephyr integration with Spotflow observability platform.
+
+if SPOTFLOW
+
+config LOG_BACKEND_SPOTFLOW
+	bool "Spotflow logging backend"
+	default y if SPOTFLOW
+	depends on LOG
+	depends on SPOTFLOW
+	select LOG_OUTPUT
+	select ZCBOR
+	help
+		Enable sending logs to Spotflow cloud.
+
+config MBEDTLS_SSL_MAX_CONTENT_LEN
+	int
+	default 4096 if SPOTFLOW
+
+config MBEDTLS_MPI_MAX_SIZE
+	int
+	default 512 if SPOTFLOW && NRF_SECURITY
+
+config MAIN_STACK_SIZE
+	default 4096 if SPOTFLOW
+
+config SPOTFLOW_SERVER_HOSTNAME
+	string
+	depends on SPOTFLOW
+	default "mqtt.spotflow.io"
+
+config SPOTFLOW_SERVER_PORT
+	int
+	depends on SPOTFLOW
+	default 8883
+
+if LOG_BACKEND_SPOTFLOW
+
+config SPOTFLOW_DEVICE_ID
+	string "Device ID device will use to connect to Spotflow observability platform"
+	help
+		Device ID is a unique identifier for
+		your device in Spotflow Observability platform
+
+config SPOTFLOW_INGEST_KEY
+	string "Password device will use to connect to Spotflow observability platform"
+	help
+		Ingest key is a secret key used to authenticate your device
+		It can be found in Spotflow Observability platform
+
+config HEAP_MEM_POOL_ADD_SIZE_SPOTFLOW
+	int "Additional heap size required by Spotflow"
+	default 6144 if LOG_BACKEND_SPOTFLOW
+	help
+		Additional heap size required by Spotflow logging backend.
+		Increase this value if you experience issues with logs being sent.
+
+config SPOTFLOW_LOG_BACKEND_QUEUE_SIZE
+	int "Size of Spotflow log backend queue"
+	default 64
+	help
+		Size of the queue used by Spotflow logging backend to store logs
+		before sending them to the server.
+		Increase this value if you experience issues with logs being dropped.
+
+config SPOTFLOW_LOG_BUFFER_SIZE
+	int "Size of Spotflow log buffer"
+	default 512
+	help
+		Size of the buffer used by Spotflow logging backend to store logs before serialization.
+		Make this at least as big as your longest expected log line.
+		Increase this value if you experience issues with logs being dropped because of buffer overflow.
+
+config SPOTFLOW_CBOR_LOG_MAX_LEN
+	int "Size of Spotflow CBOR log buffer"
+	default 1024
+	help
+		Size of the buffer used by Spotflow logging backend to serialize the logs to CBOR format.
+		Increase this value if you experience issues with logs being dropped because of buffer overflow.
+
+config SPOTFLOW_MQTT_LOG_THREAD_CUSTOM_PRIORITY
+	bool "Use custom priority for MQTT log thread"
+	default n
+	help
+		Enable this option to use custom priority for the thread used to send logs to Spotflow server.
+		If disabled, the thread will use default K_LOWEST_APPLICATION_THREAD_PRIO.
+
+config SPOTFLOW_MQTT_LOG_THREAD_PRIORITY
+	int "MQTT log thread priority"
+	default 14
+	help
+		Priority of the thread used to send logs to Spotflow server.
+		Increase this value if you experience issues with logs being sent.
+		Defaults to default K_LOWEST_APPLICATION_THREAD_PRIO
+
+config SPOTFLOW_PROCESSING_THREAD_STACK_SIZE
+	int
```

To enable users to override it:

```suggestion
	int "Size of Spotflow processing thread stack"
```